jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/LocalRegistry.java

```diff
@@ -119,6 +132,37 @@ public void stop() {
     }
   }
 
+  /** Gets local registry container IP and associates it to localhost. */
+  private String getAndMapRegistryContainerIp() {
+    String containerIp;
+
+    // Gets local registry container IP
+    List<String> dockerTokens =
+        Lists.newArrayList(
+            "docker",
+            "inspect",
+            "-f",
+            "'{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}'",
+            containerName);
+    try {
+      String result = new Command(dockerTokens).run();
+      // Remove single quotes and LF from result (e.g. '127.0.0.1'\n)
+      containerIp = result.replaceAll("['\n]", "");
+    } catch (InterruptedException | IOException ex) {
+      throw new RuntimeException("Could not get local registry IP for: " + containerName, ex);
+    }
+
```

Comment:
@mpeddada1 Hmm yeah, the difference here is that `docker inspect` is called on a specific `containerName` to grab its container IP. So for each local registry instance this is not actually available after `docker run` is executed

https://github.com/GoogleContainerTools/jib/blob/dc24c0c3b132cd745e89ecc567ea7cf68d9b4066/jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/LocalRegistry.java#L76

I think the approach you mention would allow us to get the IP for the container that the build script runs in (correct me if I'm misunderstanding!). I haven't been able to make this work yet, but something like:
```
export DOCKER_IP="$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $(hostname))"
``` 
There may still be nuances between the VM-based / container-based build environments, but I'll experiment more with the two; if there could be a fixed IP replacement, that would definitely make the test refactoring a lot more straightforward!

Reply:
I see, thanks for the context. Using the docker IP as calculated here was working for other cases but just not for web-based application so this shouldn't be a concern atm. Since the web apps ping localhost:8080, I'm curious if there is an issue with port 8080 being unreachable? I think  `sudo lsof -i :8080`  can be used to verify this 
(1) https://unix.stackexchange.com/questions/62247/how-do-i-know-what-service-is-running-on-a-particular-port-in-linux
(2) https://stackoverflow.com/questions/48352370/curl-7-couldnt-connect-to-host-what-is-wrong-in-my-settings

Reply:
Oh that is a good point! Confirmed that the port is unallocated, but came across something that might help narrow down the problem: when splitting up the steps from `WarCommandTest.testWar_jetty` and executing them individually, I was able to retrieve the contents okay with `curl` command.

https://github.com/GoogleContainerTools/jib/blob/9f2d0ba16dfb02eece1b40f55c6e2f1c42c7e178/jib-cli/src/integration-test/java/com/google/cloud/tools/jib/cli/WarCommandTest.java#L82

```
cd ~/jib/jib-cli/src/integration-test/resources/warTest && ./gradlew clean build war
cd ~/jib && ./gradlew jib-cli:installDist

./jib-cli/build/install/jib/bin/jib war --target docker://exploded-war jib-cli/src/integration-test/resources/warTest/build/libs/standard-war.war

docker run --rm --detach -p8080:8080 exploded-war --network=host
curl http://localhost:8080/ (curl http://<container-ip>:8080/ also worked)
```
However running the test individually still fails - this leads me to suspect the incompatibility lies in `HttpURLConnection` and the setup in `getContent(URL url)`helpers, rather than earlier build/run steps - going to dig in this direction a bit more!

https://github.com/GoogleContainerTools/jib/blob/9f2d0ba16dfb02eece1b40f55c6e2f1c42c7e178/jib-cli/src/integration-test/java/com/google/cloud/tools/jib/cli/WarCommandTest.java#L160

Reply:
@mpeddada1 Looping back on the use of container IP, I've reverted most of the code changes from this in [858accc](https://github.com/GoogleContainerTools/jib/pull/3886/commits/858accca9accb1db1f8066d06fcef1474e5fc6a7) given the newly suggested VM IP workaround, hopefully this is looking like a much cleaner solution now!   

Having this VM IP may also help to unblock the web app tests - I suspect the tests’ usage of `java.net.HttpUrlConnection` is failing on the first localhost resolution attempt, and does not continue to the entry added at the end of `/etc/hosts`, unlike when making a CURL request to the running container (which was able to eventually succeed). ([stackoverflow ref]( https://stackoverflow.com/questions/7478424/java-httpurlconnection-dns-resolution-with-multiple-ip-addresses))

```
curl -v localhost:8080/hello
*   Trying 127.0.0.1:8080...
* TCP_NODELAY set
* connect to 127.0.0.1 port 8080 failed: Connection refused
*   Trying 192.168.9.1:8080...
* TCP_NODELAY set
* Connected to localhost (192.168.9.1) port 8080 (#0)

```
I'm not sure why my attempts to use `[container-ip]:8080` also ran into issues earlier, but was just able to get `[vm-ip]:8080` to succeed locally for a few of these tests - going to try applying this more broadly and see if it can resolve the outstanding errors 🤞

Reply:
That's awesome news @emmileaf! That's great to know that using the VM IP ended up doing the trick. As a quick reminder, before we are ready to merge the PR, it may also be worth verifying this behavior with the continuous job since the presubmits only run a subset of tests.